### DISCUSSION TITLE
PN code sequence generator and detector

### DISCUSSION THREAD

**Discussion by xueweiwujxw** [SpinalHDL/SpinalHDL#1617] [Ideas]
Hi, all.
I want to share [PN code sequence generator and detector](https://github.com/xueweiwujxw/SpinalUtils/tree/dev/src/main/scala/spinalutils/pn).
Is it necessary to merge this code into SpinalHDL?
And where should I put this code into SpinalHDL

**Comment by Dolu1990** [comment]
Hi,

> Is it necessary to merge this code into SpinalHDL?

It isn't necessary, you can instead create repo as https://github.com/SpinalHDL/SpinalCrypto for instance.
The you can include your project as a library into another project

**Reply by xueweiwujxw** [reply]
Ok, thank you for your advise.
I will commit code to this repository [xueweiwujxw/SpinalUtils](https://github.com/xueweiwujxw/SpinalUtils)
Can I use this name `SpinalUtils` if I want to publish my code to maven?

**Reply by Dolu1990** [reply]
> Can I use this name SpinalUtils if I want to publish my code to maven?

Sure no worries, especialy as you set ThisBuild / organization := "com.github.xueweiwujxw" :)

**Comment by mrberman87** [comment]
@xueweiwujxw You may consider posting it to https://github.com/fayalalebrun/awesome-spinalhdl so people can find it associated with other user utility repos.

**Reply by xueweiwujxw** [reply]
Ok, I would try when I publish a stable version.
Thank you.
